src/components/actions/throttle.ts

```diff
@@ -19,10 +19,14 @@
  */
 const THROTTLE_QUEUE_THRESHOLD = 200 // ms
 
-type EventObject<T, U> = { name: string; data: T; action: (data: T) => Promise<U> | U }
+interface EventObject<T> {
+  name: string
+  data: T
+  action: (data: T) => unknown
+}
 
-export class EventThrottle<T, U> {
-  queue: EventObject<T, U>[]
+export class EventThrottle {
+  queue: EventObject<any>[]
```

Comment:
Okay, this has to be a truly legit use of `any`. I wanted to keep the Generic on the "EventObject", to enforce that the input "data" has to be consistent with the "action" function, but I *don't* want to have to explicitly specify a type when creating the "Event Throttle" object, as it should handle any type of Event object.

But, annoyingly, it requires a type in this definition, so it has to be `any`, right?

See [this](https://www.typescriptlang.org/play/?jsx=0#code/PTAEHUFNQCwQwG7TgO1AewEYCtIGMAXUAgTwAdoB3ASwJmMvVDICd0KWDrIBnAGgBQIUKgAmoSlVRECTSCgBm6Fnmh04RTOjrM2HLrxEto6BcRjQecALZrykQcMwBXIpNCj0KAORE8cYwl4GXs+UB4md2ceZFAAcXlIFmo8ADpQAAVjAlJwgjEA8R4CZwUFVIFqFAIkhThVUAARDTgAeRx8AgAeABUAPlAAbwFQXXYk0gBGAC5QHpGx-RIAJln5gF8BITAAIVdzeQloeCQRFG0LFgwOwmJQoJT6fzQYdAAbcVQjFjgSDDM6JBtk0Wu1cLcsOCCDx0lAPF5fBJpMQmMYAI7OaiBdQyCwg-Jgzo8UCYZBvN7-A7Aqy2O4UMIRWDoSgSdDOD5HeHmajEmh0NlEABEqBIgoA-JVqrV6tAAMJefJVJKE27DUYoGyQWbFZIoADmC1ELQAkjVrDxZs0CTduiK+gBtAC6AnWQA) for a simplified example of the case.

Reply:
Could do something like this: https://github.com/msupply-foundation/conforma-server/compare/1120-fix-trigger-block-for-review-stats-update...1120-fix-trigger-block-for-review-stats-update(No-any-example)

Reply:
Yeah, except I don't really like having to wrap the `action` into an additional anonymous function call just to keep Typescript happy. I like being able to pass the `action` and `data` in separately and calling `action` on `data` directly, and using the Generic to enforce consistency between action method and its input.

Minor thing, slightly prefer my way. (Plus, any thoughts about the even more general case in the [playground example](https://www.typescriptlang.org/play/?jsx=0#code/PTAEHUFNQCwQwG7TgO1AewEYCtIGMAXUAgTwAdoB3ASwJmMvVDICd0KWDrIBnAGgBQIUKgAmoSlVRECTSCgBm6Fnmh04RTOjrM2HLrxEto6BcRjQecALZrykQcMwBXIpNCj0KAORE8cYwl4GXs+UB4md2ceZFAAcXlIFmo8ADpQAAVjAlJwgjEA8R4CZwUFVIFqFAIkhThVUAARDTgAeRx8AgAeABUAPlAAbwFQXXYk0gBGAC5QHpGx-RIAJln5gF8BITAAIVdzeQloeCQRFG0LFgwOwmJQoJT6fzQYdAAbcVQjFjgSDDM6JBtk0Wu1cLcsOCCDx0lAPF5fBJpMQmMYAI7OaiBdQyCwg-Jgzo8UCYZBvN7-A7Aqy2O4UMIRWDoSgSdDOD5HeHmajEmh0NlEABEqBIgoA-JVqrV6tAAMJefJVJKE27DUYoGyQWbFZIoADmC1ELQAkjVrDxZs0CTduiK+gBtAC6AnWQA)?)